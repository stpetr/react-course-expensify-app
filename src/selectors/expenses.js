import moment from 'moment'

export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const createdAt = moment(expense.createdAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdAt, 'day') : true
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdAt, 'day') : true
        const textMatch = typeof text === 'string' ? expense.description.toLowerCase().includes(text.toLocaleLowerCase()) : true

        return startDateMatch && endDateMatch && textMatch
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? -1 : 1
        }

        if (sortBy === 'amount') {
            return a.amount < b.amount ? -1 : 1
        }
    })
}