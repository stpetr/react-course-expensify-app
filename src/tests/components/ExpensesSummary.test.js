import React from 'react'
import { shallow } from 'enzyme'
import ExpenseSummary, { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'
import filters from '../fixtures/filters'

test('should render expenses summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={10500} />)
    expect(wrapper).toMatchSnapshot()
})

test('should render expenses summary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={8} expenseTotal={168050} />)
    expect(wrapper).toMatchSnapshot()
})