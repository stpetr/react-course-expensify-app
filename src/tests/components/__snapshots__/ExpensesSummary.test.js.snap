// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    8
     
    expenses
     totalling 
    $1,680.50
    .
  </p>
</div>
`;

exports[`should render expenses summary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $105.00
    .
  </p>
</div>
`;
